### PR TITLE
Eba template

### PR DESCRIPTION
**Pull request by canonici**
Update of the eba template to make it responsive and a little bit more _"stylish"_ .

### CODE REVIEW

File: src/eba_tools.eliom [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+[%%shared
+ module NavigationBar : sig
```
Comment (vasilisp) [inline src/eba_tools.eliom]: We are not really into CamelCase :), I would much prefer `Navigation_bar`.

File: src/eba_tools.eliom [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+   ) Eliom_service.t
+
+   type elt = string * service
+
+   let li_of_elt elt = Eliom_content.Html.F.(
+     let text, service = elt in
+     li [a ~service [pcdata text] ()]
+   )
+
+   let of_elt_list ?(elt_class = []) elt_list = Eliom_content.Html.F.(
```
Comment (vasilisp) [inline src/eba_tools.eliom]: `ul_of_elt_list` to be consistent with `li_of_elt`? Or maybe `ul_of_elts`?

File: src/eba_tools.eliom [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+     @@ List.map li_of_elt elt_list
+   )
+
+ end
+]
+
+
+let%shared popup_button
+    ~button_name
+    ?(button_class = ["eba_popup_button"])
```
Comment (vasilisp) [inline src/eba_tools.eliom]: Dashes are way more common for class names, no?

File: template.distillery/PROJECT_NAME_container.eliom [hunk 4]
```diff
@@ -3,62 +3,58 @@
...
+    ("About",%%%MODULE_NAME%%%_services.about_service);
+    ("Demo",%%%MODULE_NAME%%%_services.otdemo_service)
+  ] in
+  fun () ->
+    Eba_tools.NavigationBar.of_elt_list
+      ~elt_class:["nav";"navbar-nav"]
+      nav_elts
 
-let header ?user () =
+let%shared eba_header ?user () = Eliom_content.Html.F.(
```
Comment (vasilisp) [inline template.distillery/PROJECT_NAME_container.eliom]: Do we need the `eba_` prefix?

**Comment by vasilisp** [comment]
I have no comments other than the minor comments above. It certainly looks better than what we had before :).

File: template.distillery/PROJECT_NAME_container.eliom (hunk 4 above)
Comment (canonici) [inline template.distillery/PROJECT_NAME_container.eliom]: There is already the header function in Eliom_content.Html modules so it prevents potential collision.

File: src/eba_tools.eliom [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+     Eliom_service.get,
+     Eliom_service.att,
+     Eliom_service.non_co,
+     Eliom_service.non_ext,
+     Eliom_service.reg,
+     [ `WithoutSuffix ],
+     unit,
+     unit,
+     Eliom_service.non_ocaml
+   ) Eliom_service.t
```
Comment (vasilisp) [inline src/eba_tools.eliom]: The type constraints are quite strong. Maybe relax them a bit (with a polymorphic `service` type, or with no named type at all)? For reference, look at the signature of `a` in Eliom: https://github.com/ocsigen/eliom/blob/master/src/lib/eliom_form_sigs.shared.mli#L191

File: template.distillery/PROJECT_NAME_otdemo.eliom [hunk 6]
```diff
@@ -58,78 +69,264 @@ end
...
-    let prev = Ot_carousel.previous
-      ~a:[a_class ["button"]]
-      ~change:[%client carousel_change]
-      ~pos
-      [pcdata "←"]
+    let carousel,
+      Some ribbon,
+      Some prev,
+      Some next,
+      Some bullets
```
Comment (vasilisp) [inline template.distillery/PROJECT_NAME_otdemo.eliom]: Can you make this pattern matching exhaustive? I think we just need a `failwith` with a reasonable message for any case than `Some _, Some _, Some _, Some _`.

File: template.distillery/static!css!eba.css [hunk 7]
```diff
@@ -1,228 +1,319 @@
...
+a:hover {
+    cursor: pointer;
 }
 
-input {
-  border: none;
-  padding: 2px 4px;
+html {
+    box-sizing: border-box;
+    font-familiy: sans-serif;
```
Comment (vasilisp) [inline template.distillery/static!css!eba.css]: typo: `font-family`

File: template.distillery/static!css!eba.css [hunk 8]
```diff
@@ -1,228 +1,319 @@
...
+/******************************************************************************/
+/*********************Default style for some basic elements********************/
+/******************************************************************************/
+
+hr {
+    color: transparent;
+}
+
+br {
+    colot: transparent;
```
Comment (vasilisp) [inline template.distillery/static!css!eba.css]: typo: `color`

**Comment by canonici** [comment]
We need to merge #97 and the next pr I'll do for multimail support (which also needs #97) before this one.

**Comment by dannywillems** [comment]
When will it be merged? I want to switch to ocsigen-start but I always work with the template eba.pgocaml.

**Comment by canonici** [comment]
There are some features in the template that needs #99 to be merged first.
You can create a new branch based on ocsigen-start master and pull the necessary changes:
git checkout master
git checkout -b mynewbranch
git pull http://github.com/canonici/eliom-base-app.git#multimail
git pull http://github.com/canonici/eliom-base-app.git#eba_template

Then you should have a working branch with multimail and the template.

Or you can pull from my branch where everything is already merged (it also includes the changes of #80 ) :
git checkout master
git checkout -b mynewbranch
git pull http://github.com/canonici/eliom-base-app.git#test_os

**Comment by dannywillems** [comment]
@canonici So when #99 will be merged, we can also merge this branch?

**Comment by canonici** [comment]
Yes we can.

**Comment by vasilisp** [comment]
This is safe because it only touches the template, and it is better if any work (e.g., by @dannywillems) builds on the new template. Merging!

**Pull request merged**